File: kotlin-pr/50/kotlin/code-quality-naming/DataModel.kt
class DataModel {
    // {fact rule=code-quality-naming@v1.0 defects=1}
    private var d: MutableList<String> = mutableListOf()
    private var c: Int = 0
    private var f: Boolean = false

    fun p(arr: List<String>) {
        d.clear()
        c = 0
        f = false

        arr.forEach { i ->
            if (i.isNotEmpty()) {
                d.add(i)
                c++
            }
        }

        f = c > 0
    }
    // {/fact}
}

File: ruby-pr/53/ruby/code-quality-naming/data_model.rb
class DataModel
  # {fact rule=code-quality-naming@v1.0 defects=1}
  def initialize
    @d = []
    @c = 0
    @f = false
  end

  def p(arr)
    @d = []
    @c = 0
    @f = false

    arr.each do |i|
      unless i.empty?
        @d << i
        @c += 1
      end
    end

    @f = @c > 0
  end
  # {/fact}
end